src/open_producten/producttypen/tests/test_load_upn.py

```diff
@@ -22,80 +31,124 @@ def call_command(self, *args, **kwargs):
         )
         return out.getvalue()
 
-    def test_load_csv_with_correct_columns(self):
-        file_name = "data/upl.csv"
-        path = os.path.join(os.path.dirname(__file__), file_name)
-        result = self.call_command(path)
-        self.assertEqual(result, f"Importing upn from {path}...\nDone (1 objects).\n")
+    def test_call_command_without_file_and_url(self):
+        with self.assertRaisesMessage(
+            Exception, "Either --file or --url must be specified."
+        ):
+            self.call_command()
+
+    def test_call_command_wit_file_and_url(self):
+        with self.assertRaisesMessage(
+            Exception, "Only one of --file or --url can be specified."
+        ):
+            self.call_command("--file", self.path, "--url", "https://example.com")
+
+    def test_with_other_file_extension(self):
+        path = os.path.join(TESTS_DIR, "data/upl.txt")
+        with self.assertRaisesMessage(Exception, "File format is not csv."):
+            self.call_command("--file", path)
+
+    def test_with_csv_file(self):
+        result = self.call_command("--file", self.path)
+        self.assertEqual(
+            result,
+            f"Importing upn from {self.path}...\nDone\nCreated 1 product names.\nUpdated 0 product names.\n0 product names did not exist in the csv.\n",
+        )
+
         self.assertEqual(UniformeProductNaam.objects.count(), 1)
         self.assertEqual(
-            UniformeProductNaam.objects.first().naam, "aangifte vertrek buitenland"
+            UniformeProductNaam.objects.get().naam, "aangifte vertrek buitenland"
         )
         self.assertEqual(
-            UniformeProductNaam.objects.first().uri,
+            UniformeProductNaam.objects.get().uri,
             "http://standaarden.overheid.nl/owms/terms/AangifteVertrekBuitenland",
         )
 
-    def test_load_csv_with_incorrect_columns(self):
-        file_name = "data/wrong-upl.csv"
-        path = os.path.join(os.path.dirname(__file__), file_name)
+    def test_with_csv_url_404(self):
+        self.requests_mock.get(
+            status_code=404,
+            url="https://test/upl.csv",
+        )
 
-        with self.assertRaisesMessage(CommandError, "'URI' does not exist in csv"):
-            self.call_command(path)
+        with self.assertRaisesMessage(Exception, "Url returned status code: 404"):
```

Comment:
Minor complaint, but this should capture the specific exception. Also it should maybe raise a command error if it no longer does